data/pigui/libs/wrappers.lua

```diff
@@ -0,0 +1,758 @@
+--
+--   nil
+--
+function ui.popup(name, fun)
+	if pigui.BeginPopup(name) then
+		fun()
+		pigui.EndPopup()
+	end
+end
+
+--
+-- Function: ui.child
+--
+-- ui.child(id, size, flags, fun)
+--
+-- Define a child window
+--
+--
+-- Example:
+--
+-- > 
+--
+-- Parameters:
+--
+--   id    - String, a unique name for the window,
+--           used to group its children
+--   size  - (Optional)Vector2
+--   flags - (Optional)Table, options:
+--              - NoTitleBar                : Disable title-bar
+--              - NoResize                  : Disable user resizing with the lower-right grip
+--              - NoMove                    : Disable user moving the window
+--              - NoScrollbar               : Disable scrollbars (window can still scroll with
+--                                            mouse or programmatically)
+--              - NoScrollWithMouse         : Disable user vertically scrolling with mouse wheel.
+--                                            On child window, mouse wheel will be forwarded to
+--                                            the parent unless NoScrollbar is also set.
+--              - NoCollapse                : Disable user collapsing window by double-clicking
+--                                            on it
+--              - AlwaysAutoResize          : Resize every window to its content every frame
+--              - NoSavedSettings           : Never load/save settings in .ini file
+--              - NoInputs                  :
+--              - MenuBar                   : Has a menu-bar
+--              - HorizontalScrollbar       : Allow horizontal scrollbar to appear (off by default).
+--              - NoFocusOnAppearing        : Disable taking focus when transitioning from hidden to
+--                                            visible state
+--              - NoBringToFrontOnFocus     : Disable bringing window to front when taking focus
+--                                            (e.g. clicking on it or programmatically giving it
+--                                            focus)
+--              - AlwaysVerticalScrollbar   : Always show vertical scrollbar
+--              - AlwaysHorizontalScrollbar : Always show horizontal scrollbar
+--              - AlwaysUseWindowPadding    : Ensure child windows without border uses
+--                                            style.WindowPadding (ignored by default for
+--                                            non-bordered child windows, because more convenient)
+--   fun   - Function, a function that is called to define the popup contents
+--
+-- Returns:
+--
+--   nil
+--
+function ui.child(id, size, flags, fun)
+	if flags == nil and fun == nil then -- size is optional
+		fun = size
+		size = Vector2(-1,-1)
+		flags = {}
+	elseif fun == nil then
+		fun = flags
+		flags = {}
+	end
+
+	pigui.BeginChild(id, size, flags)
+	fun()
+	pigui.EndChild()
+end
+
+function ui.registerHandler(name, fun)
+	pigui.handlers[name] = fun
+end
```

Comment:
This function should ideally go in `baseui.lua`.

Reply:
Moved to baseui.lua with `ui.registerTheme`.